File: financial-domain-common/src/main/java/com/nixsolutions/financial/domain/enums/Endpoints.java
package com.nixsolutions.financial.domain.enums;

public enum Endpoints implements EnumWithValue<Endpoints, String>
{
  TIME_SERIES_INTRADAY("TIME_SERIES_INTRADAY"),
  TIME_SERIES_DAILY("TIME_SERIES_DAILY"),
  TIME_SERIES_DAILY_ADJUSTED("TIME_SERIES_DAILY_ADJUSTED"),
  TIME_SERIES_WEEKLY("TIME_SERIES_WEEKLY"),
  TIME_SERIES_MONTHLY("TIME_SERIES_MONTHLY");

  private String value;

  Endpoints(String value)
  {
    this.value = value;
  }

  public String getValue()
  {
    return value;
  }

  @Override
  public Endpoints getByValue(String value)
  {
    return iterateThroughValuesForMatch(value, values());
  }
}

File: financial-domain-starter/src/main/kotlin/com/nixsolutions/financial/domain/enums/Endpoints.kt
package com.nixsolutions.financial.domain.enums

enum class Endpoints private constructor(override val value: String) : EnumWithValue<Endpoints, String> {
  TIME_SERIES_INTRADAY("TIME_SERIES_INTRADAY"),
  TIME_SERIES_DAILY("TIME_SERIES_DAILY"),
  TIME_SERIES_DAILY_ADJUSTED("TIME_SERIES_DAILY_ADJUSTED"),
  TIME_SERIES_WEEKLY("TIME_SERIES_WEEKLY"),
  TIME_SERIES_MONTHLY("TIME_SERIES_MONTHLY");

  override fun getByValue(value: String): Endpoints {
    return iterateThroughValuesForMatch(value, values())
  }
}
